### DISCUSSION TITLE
Ray doesn't seem to work from a docker-image

### DISCUSSION THREAD

**Discussion by robertgdev** [spatie/ray#672] [Q&A] [answered]
**Describe the bug**
Ray works on native Linux (to verify that it works at all) but when attempting to use it from a running docker-image, it doesn't seem to send any data to the running ray app (OK, it might be sending something, but no output ever arrives on the actual ray app).  

**Versions**
2.0.1

You can use `composer show` to get the version numbers of:
- spatie/ray package version: 1.34.1
- spatie/laravel-ray package version (if applicable): N/A

PHP version: 8.1.3
Laravel version (if applicable): N/A

**To Reproduce**
1. Added a ```ray("blah")``` call at the start of my controller file

**Expected behavior**
I expect the "blah" to show up in the ray app, but nothing happens. 

**Screenshots**
N/A ... nothing happens, so no screenshot to show

**Desktop (please complete the following information):**
 - OS: [e.g. macOS]: Ubuntu Linux (running KDE desktop)
 - Version [e.g. 22]: 21.10
 - App running from a docker image (PHP 8.,1.3)
 
 **Additional context**
I followed the steps in your docs: 
1) ```composer require spatie/ray```
2) Added ray.php file (from your 'docker' config page) to project (tried a number of directories, this made no difference)
3) Fixed paths in the ray.php file for localhost + docker environment
4) Added ```127.0.0.1 host.docker.internal``` (actually, this was already there)
5) Added the 'extra_hosts" entries in 2 places (to make sure, I'm no docker expert) to docker-composer.yml
6) Added a call to ```ray("blah")``` to the start of my controller file
7) To make sure that it gets executed (don't ask, our app is a legacy app which is a bit over-convoluted) I changed the ray() call to ```rayxxx("blah")``` (to ensure that it gets executed) and promptly received an error 500. So the call to ray is being executed. 
8) Noticed that the sample docker config file had no ```'enable' => true``` entry, so I added this -> still no success
9) Check the ports numbers in ray.php and on the desktop app and they match (23517)

Please advise. Thank you.

**Comment by freekmurze** [comment]
We don't have any extra information besides what's in the docs. The most important step is to make sure those ports match up. I hope other docker users can help you with this one.

**Reply by robertgdev** [reply]
Thank you for getting back to me. If anybody out there knows how to get this to work with docker, I'd very much appreciate some hints/tips ...

**Comment by robertgdev** [comment, marked as answer]
Figured it out: I had ufw (the Ubuntu firewall) enabled to block any non-http ports and didn't realize that this would disable connections from the docker image. Disabling the firewall solved the issue (you could also just open the relevant ports but I'm too lazy to test this now).

**Reply by lode** [reply]
For future reference, some more info was added in https://github.com/spatie/myray.app/pull/33.
